rootex/script/script.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "component.h"
+#include "event_manager.h"
+#include "script/interpreter.h"
+#include "entity.h"
+
+class LuaTextResourceFile;
+
+class Script
+{
+private:
+	sol::environment m_ScriptEnvironment;
+	String m_ScriptFile;
+	HashMap<String, String> m_Overrides;
+
+	friend class Entity;
```

Comment:
Doesn't need to be a friend. `Entity` only ever tried to access `Script::m_ScriptEnvironment` which is not worth making them friends. Make setters and getters and use them instead to set/get the env